pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -188,6 +208,183 @@ func handleInferenceGraphRawDeployment(ctx context.Context, cl client.Client, cl
+				isPresent = true
+				break
+			}
+		}
+		if !isPresent {
+			clusterRoleBinding.Subjects = append(clusterRoleBinding.Subjects, rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      saName,
+				Namespace: graph.GetNamespace(),
+			})
+			_, err = clientset.RbacV1().ClusterRoleBindings().Update(ctx, clusterRoleBinding, metav1.UpdateOptions{FieldManager: InferenceGraphControllerName})
+			if err != nil {
+				return errors.Wrapf(err, "fails to bind privileges for auth verification to inference graph")
+			}
+		}
+	}
+
+	return nil
+}
+
+func removeAuthPrivilegesFromGraphServiceAccount(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1.InferenceGraph) error {
+	clusterRole, err := clientset.RbacV1().ClusterRoleBindings().Get(ctx, constants.InferenceGraphAuthCRBName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "fails to get cluster role binding kserve-inferencegraph-auth-verifiers while deconfiguring inference graph auth")
+	}
+
+	isPresent := false
+	saName := graph.ResolveServiceAccountName()
+	for idx, subject := range clusterRole.Subjects {
+		if subject.Kind == "ServiceAccount" && subject.Name == saName && subject.Namespace == graph.GetNamespace() {
+			isPresent = true
+
+			// Remove the no longer needed entry
+			clusterRole.Subjects[idx] = clusterRole.Subjects[len(clusterRole.Subjects)-1]
+			clusterRole.Subjects = clusterRole.Subjects[:len(clusterRole.Subjects)-1]
+			break
+		}
+	}
+
+	if isPresent {
+		_, err = clientset.RbacV1().ClusterRoleBindings().Update(ctx, clusterRole, metav1.UpdateOptions{FieldManager: InferenceGraphControllerName})
+		if err != nil {
+			return errors.Wrapf(err, "fails to remove privileges for auth verification from inference graph")
+		}
+	}
+
+	return nil
+}
+
+func checkIfGraphServiceAccountIsOwned(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1.InferenceGraph) (*corev1.ServiceAccount, bool, error) {
+	// Fetch the ServiceAccount, if it would exist
+	saName := graph.ResolveServiceAccountName()
+	existingServiceAccount, err := clientset.CoreV1().ServiceAccounts(graph.GetNamespace()).Get(ctx, saName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, false, nil
+		}
+		return nil, false, errors.Wrap(err, "fails to get inference graph service account")
+	}
+
+	// Check if the SA is owned by the IG.
+	// TODO: For now, do a custom check. After updating to controller-runtime v0.20, change the check to use HasOwnerReference util func.
+	isOwner := false
+	saOwnerReference := existingServiceAccount.GetOwnerReferences()
+	if len(saOwnerReference) > 0 {
+		isOwner = saOwnerReference[0].APIVersion == graph.APIVersion &&
+			saOwnerReference[0].Kind == graph.Kind &&
+			saOwnerReference[0].Name == graph.GetName() &&
+			saOwnerReference[0].UID == graph.UID
```

Comment:
Should we check all owner refs as there could be more than one in some cases and ours is not necessarily the "first" one to make this check more robust?

Reply:
May we create an issue to now forget about this? :)


Is there a test case for custom SAs?

Reply:
> Should we check all owner refs as there could be more than one in some cases and ours is not necessarily the "first" one to make this check more robust?

@pierDipi my thinking is that when kserve-controller creates the SA inside `handleInferenceGraphRawAuthResources`, for the time being we would expect the InferenceGraph to be the only owner (until some other controller starts operating on the SA). Still, we can make it more robust if you think it is worth it.

BTW, once the upgrade to controller-runtime v0.20 can happen, we can remove our own implementation and the `HasOwnerReference` func would cover that case.

> Is there a test case for custom SAs?

Hmm...  @spolti  I think no. I'll add some.